daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -738,6 +738,11 @@ private[validation] object Typing {
       TOptional(TTyCon(tpl))
     }
 
+    private def typeOfTyCon(tyCon: TypeConName): Type = {
+      lookupDataType(ctx, tyCon)
```

Comment:
This only checks that the type exists but it could be a type with parameters in which case `SBTyCon` seems like a weird primitive since it can’t actually be used to get a textual representation of a type but only the type constructor.

I would suggest to either limit this to types with no type parameters or even to just template types. I don’t really have a preference here, what do you think @remyhaemmerle-da?

Reply:
The first one seems like the most consistent. Since builtin types (e.g. `Int64`) are not supported either.

Reply:
Please go ahead with  your prefer choice.